### PR TITLE
Overview: add 'Active' SSL status for managed subdomains

### PR DESCRIPTION
**Pull request by chad1008**
Related to 8770-gh-Automattic/dotcom-forge

~Dependent on D158944-code~ This diff is no longer needed!

## Proposed Changes

- Updates the domains table to display `Active` SSL status for wpcom managed subdomains like `*.wordpress.com` or `*.wpcomstaging.com`.
- Cleans up the rendering logic of the `DomainsTableSslCell` component a bit
- Updates the casing of SSL in variable names from the previous PR for this feature, for greater consistency

**Before**
image

**After**
image

## Why are these changes being made?

After adding SSL status to the Domains Table, we decided rendering a null value (`-`) for managed subdomains serves to hide the fact that these subdomains are secured by the root domain's wildcart certificate. By revealing that these subdomains are covered by an SSL certificate, we highlight and security of those default subdomains.

We're handling this on the front end (using a new property introduced in D158944-code). That value is populated by a function that checks the subdomain in question against a list of a8c owned domains with wildcard SSL certs. Effectively, if it's a subdomain on one of those URLs, we can be confident it's covered.

I didn't use that function to modify the value of the existing `ssl_status` property because that prop has historically been used for registered domains with their own certificates, and I didn't want to pollute that value and potentially introduce unexpected behavior elsewhere.

## Testing Instructions
**Atomic**
- Visit `/overview` for an atomic site with a custom domain name attached to it
- Confirm that the `*.wpcomstaging.com` subdomain is showing that it has an Active SSL certificate
- Confirm that the custom domain still also shows an active SSL certificate
- Confirm that the word `Active` is a link for the custom domain, opening the domain details page with the SSL section expanded
- Confirm the word `Active` is not a link (it will render as a `span`) for the free subdomain

**Simple**
- Visit `/overview` for a simple site
- Confirm that the `*.wordpress.com` subdomain is showing that it has an Active SSL certificate
- Confirm that the word `Active` is not a link

### CODE REVIEW

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (chad1008) [inline packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx:25]: For registered domains we render a link to domain details with the SSL section already expanded. Because subdomains don't actually have a certificate of their own, we don't want a link, so we render a span instead.

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (chad1008) [inline packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx]: Registered domains with their own certs render a link to view domain and certificate details. Because that isn't an option for subdomains, we'll render a `span` instead.

With this new logic, the `-` we rendered previously should never be reached, but I've left it in as a fallback. If that return value ever is reached, it'll look better if it isn't blank and matches the dashes shown elsewhere on the table.

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (fredrikekelund) [inline packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx:13]: Either way is fine by me, but this does read a bit more calmly, I guess

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (fredrikekelund) [inline packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx]: Let's avoid defining a React component inside another component and either move `Element` to the topmost scope (which would necessitate some more prop passing, but could be OK) or do something similar to the previous implementation, i.e., along these lines:

```tsx
	const buttonClassNames = clsx( 'domains-table-row__ssl-status-button', {
		[ 'domains-table-row__ssl-status-button__active' ]: isActiveSsl,
		[ 'domains-table-row__ssl-status-button__pending' ]: isPendingSsl,
		[ 'domains-table-row__ssl-status-button__disabled' ]: sslStatus === 'disabled',
	} );
	let button: React.ReactElement | string;

	if ( sslStatus ) {
		button = (
			<a
				className={ buttonClassNames }
				href={ `${ domainManagementLink }?ssl-open=true` }
				onClick={ ( event ) => event.stopPropagation() }
			>
				{ getSslStatusText() }
			</a>
		);
	} else if ( hasWpcomManagedSslCert ) {
		button = <span className={ buttonClassNames }>{ getSslStatusText() }</span>;
	} else {
		button = '-';
	}
```

FYI, defining a React component within another component isn't strictly forbidden, but it is an anti-pattern. The clearest argument against it is that it decreases performance (the inner component is redefined on every render, meaning React can't properly optimize re-renders). Performance isn't a huge concern here, it's more that I don't think there's anything really compelling us to go against the convention here.

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (fredrikekelund) [inline packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx]:
```suggestion
	const domainHasSsl = sslStatus !== null || hasWpcomManagedSslCert;
```
Nit, but we might as well be explicit about which value we're looking for here.

**Review by fredrikekelund — COMMENTED** [review]
Looking good overall and tests well 👍 Let's address my comment about the `Element` component and this should be good to go after that

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (chad1008) [reply packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx]: Good point, thanks! I feel like I've seen this pattern before but hadn't connected to dots on the poor optimization impact, thank you for elaborating!

**Comment by chad1008** [comment]
Feedback addressed @fredrikekelund. I also had to rebase to address some conflicts with `trunk`.

I pushed one [additional change](https://github.com/Automattic/wp-calypso/pull/93754/commits/c31789c95098eaffebc05b8edfb417c64580d99e) to your suggestion to avoid redeclaring those very similar classNames. It's less verbose, but also adds a little abstraction. Let me know if you think simplicity trumps brevity in this case. I can alway drop that commit before merging 🙂

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (fredrikekelund) [inline packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx]: Nit, but I would argue that verbosity is preferable here because it makes it easier to grep for those class names across the code base (i.e., if I search for `domains-table-row__ssl-status-button` across the code base, I won't find this instance).

However, I know we have several instances in `packages/domains-table/src/domains-table/style.scss` of class names being stringed together in a "similar" way (e.g., [here](https://github.com/Automattic/wp-calypso/blob/c31789c95098eaffebc05b8edfb417c64580d99e/packages/domains-table/src/domains-table/style.scss#L152)). So, there's no rule against this, but I would personally argue that it's preferable to spell out the full class name.

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (fredrikekelund) [inline packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx]:
```suggestion
		button = <span className={ getSslStatusClassnames( 'button' ) }>{ translate( 'Active' ) }</span>;
```
Nit-ish, but since we will only ever render `Active` here – I think we might as well be explicit about that.

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (fredrikekelund) [inline packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx]: Did you consider moving the modifier classes to this parent table cell, @chad1008? That would simplify the class name logic here.

**Review by fredrikekelund — COMMENTED** [review]
A few final comments before this is good to go. I would argue that moving the class names to the `<td>` element is probably the most elegant approach, but let me know what you think, @chad1008.

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (chad1008) [reply packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx]: Good point!

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (chad1008) [reply packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx]: A good point, thanks. Your next question on moving the modifier classes to the parent `td` helps a lot too, alleviating the need for a second set of modifiers.

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (chad1008) [reply packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx]: Excellent idea, thanks!

File: packages/domains-table/src/domains-table/style.scss
Comment (fredrikekelund) [inline packages/domains-table/src/domains-table/style.scss]:
```scss
	.domains-table-row__ssl-cell {
		gap: 2px;

		&__active {
			fill: var(--studio-green-50);
			color: var(--studio-green-50);
		}

		&__pending,
		&__disabled {
			fill: var(--studio-orange-50);
			color: var(--studio-orange-50);
		}
	}

	.domains-table-row__ssl-status-button {
		color: inherit;
	}
```
It looks like we can do something like this to simplify the styling here.

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (fredrikekelund) [inline packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx:46]: Optional nit, but since we aren't reusing `getSslStatusText` now, we could potentially just inline the function logic in this if clause. Up to you!

**Review by fredrikekelund — APPROVED** [review]
LGTM 👍 Apart from my comment about the CSS, this should be good to go once D158944-code has landed.

File: packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx
Comment (chad1008) [reply packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx:46]: Nice catch. I started to make this change but I actually think I prefer to keep the function as it is. The early returns ensuring only one value is rendered feels a little more resilient.

I'm trying to envision some future change where (perhaps accidentally, due to a regression elsewhere) more than one of those three conditions could be true at the same time.

It feels unlikely, but if it ever did happen somehow, we'd wind up with something like `🔒 ActivePending`, which wouldn't be great. Again, super unlikely, but I think I prefer knowing that only one value is ever possible for this cell of the table.

**Pull request merged**
